#!/usr/bin/env python
PACKAGE = "perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("gradient_detection_thld", double_t, 0, "threshold used in gradient detection", 24.0, -100.0, 100.0);
gen.add("min_sampling_distance", double_t, 0, "Minimum distance between sampling_x_start and sampling_x_end. Only used if dynamically_restrict_search_pattern == true.", 1.0, 0.5, 5);
gen.add("dynamically_restrict_search_pattern", bool_t, 0, "Wether to calculate sampling_x_end from the points provided by the lane detection or use the value specified in the parameter file", True);
gen.add("degree_of_polynomial", int_t, 0, "Degree of the polynomial which is fitted to the points provided by the lane detection", 2, 2, 10);
gen.add("trim_factor", double_t, 0, "Used to calculate the trimmed end points of each scan line", 0.1, 0, 0.5);

exit(gen.generate(PACKAGE, "perception", "RoadWatcher"))
